File: 19th-April-Wifi_Range/python/solution.py

#  This code is written by Bhaskar


#User function Template for python3
class Solution:
    def wifiRange(self, N, S, X): 
        #code here
        check = [False] * N

        for i in range(N):
            if S[i] == '1':
                check[i: min(i + X + 1, N)] = [True] * (min(i + X + 1, N) - i)
                check[max(i - X, 0): i + 1] = [True] * (i + 1 - max(i - X, 0))
        return all(check)   
        

File: 19th-April-Wifi_Range/c++/solution.cpp
/*
  This code is written by Bhaskar
*/

//User function Template for C++

class Solution{
    public:
    bool wifiRange(int N, string S, int X){
        // code here
        bool check[N];
        memset(check, false, sizeof(check));

        for(int i=0 ; i < N ; i++){
            if(S[i] == '1'){
                memset(check + i, true, min(X + 1, N - i) * sizeof(bool));
                memset(check + max(i - X, 0) , true, X* sizeof(bool));
            }
        }

        for(bool x : check){
            if(!x){
                return false;
            }
        }
        return true;
    }
};
